template/{% if ci == 'github' %}.github{% endif %}/workflows/docs.yml.jinja

```
@@ -0,0 +1,46 @@
+name: Documentation
+
+on:
+  push:
+    branches: [main]
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  build-and-deploy:
+    if: github.ref == 'refs/heads/main'
+
+    runs-on: ubuntu-latest
+
+    environment:
+      name: github-pages
+      url: ${{ "{{" }} steps.deployment.outputs.page_url {{ "}}" }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      
+      - name: Install uv
+        uses: astral-sh/setup-uv@v6
+        
+      - name: Build docs
+        run: uv run mkdocs build
+      
+      - name: Configure GitHub Pages
+        uses: actions/configure-pages@v5
+        
+      - name: Upload GitHub Pages artifact 
+        uses: actions/upload-pages-artifact@v3
+        with:
+          path: site
+          
+      - name: Deploy to GitHub Pages
+        id: deployment
```

Suggested changes:
- `cancel-in-progress: true` so it cancels other jobs in the group.
- No need for `if: github.ref == 'refs/heads/main'` since it only runs on push to `main`
- `checkout@v5` is a newer version
- `upload-pages-artifact@v4` is a newer version

```suggestion
  cancel-in-progress: true

jobs:
  build-and-deploy:
    runs-on: ubuntu-latest

    environment:
      name: github-pages
      url: ${{ "{{" }} steps.deployment.outputs.page_url {{ "}}" }}

    steps:
      - name: Checkout
        uses: actions/checkout@v5
      
      - name: Install uv
        uses: astral-sh/setup-uv@v6
        
      - name: Build docs
        run: uv run mkdocs build
      
      - name: Configure GitHub Pages
        uses: actions/configure-pages@v5
        
      - name: Upload GitHub Pages artifact 
        uses: actions/upload-pages-artifact@v4
        with:
          path: site
          
      - name: Deploy to GitHub Pages
        id: deployment
```